examples/all-clusters-app/all-clusters-common/src/binding-handler.cpp

```diff
@@ -67,7 +67,8 @@ static void RegisterSwitchCommands()
 }
 #endif // defined(ENABLE_CHIP_SHELL)
 
-static void BoundDeviceChangedHandler(const EmberBindingTableEntry & binding, chip::DeviceProxy * peer_device, void * context)
+static void BoundDeviceChangedHandler(const EmberBindingTableEntry & binding, chip::Messaging::ExchangeManager * exchangeMgr,
+                                      chip::SessionHandle * sessionHandle, void * context)
```

Comment:
Is it an anti-pattern to have possibly null session handles?  I thought we wanted these to always be valid, and wanted to use SessionHolder for possibly-empty / non-existent sessions.

Reply:
(I don't feel strongly about this.  Just asking.)

Reply:
Yes, I think `chip::SessionHandle * sessionHandle` as a pointer at all is counter to our design goals and subverts the ownership model.  Suggest figuring out a way to not do this.